bert/README.md

```diff
@@ -0,0 +1,244 @@
+
+One may wonder, why start from scratch when public data isn't a concern? Granted that it is better to train on domain-specific data, can't that happen as "domain adaptation" from a pre-trained checkpoint? There are two reasons not to do this, one theoretical and one practical. The theory says that, because we are doing non-convex optimization, domain adaptation "may not be able to completely undo suboptimal initialization from the general-domain language model" [(Gu et al., 2020)](https://arxiv.org/abs/2007.15779).
+
+The practical reason is that certain outcomes are only available if the model and tokenizer are pre-trained from scratch.
+
+So, for example, if you want to use ALiBi positional embeddings (and [you probably should](https://ofir.io/The-Use-Case-for-Relative-Position-Embeddings/), they seem to improve LM perplexity, downstream accuracy, and allow the model to generalize to longer sequences than seen at train time), you need to train from scratch (or fine-tune from a checkpoint which was pre-trained with ALiBi positional embeddings, which we will be releasing!). Or if you want to use Gated Linear Units in your feedforward layers ([as recommended by Noam Shazeer](https://arxiv.org/abs/2002.05202), one of the authors of the original Transformers paper), again, you have to train with them from scratch.
+
+Another good example is domain-specific tokenization. In the biomedical domain, words may be split by the pre-trained BERT tokenizer in ways that make downstream tasks more difficult and computationally expensive. For example, the common drug "naloxone" in tokenized by `bert-base-uncased` tokenizer into the 4 tokens `[na, ##lo, ##xon, ##e]` [(Gu et al., 2020)](https://arxiv.org/abs/2007.15779), making tasks like NER more difficult and using more of the limited sequence length available.
+
+Now that we've convinced you that you should train a Mosaic BERT from scratch, let's get into the how :) 
+
+# Setup
+
+Here's what you need to get started with our BERT implementation:
+
+* Use a Docker image with PyTorch 1.12+, e.g. [MosaicML's PyTorch base image](https://hub.docker.com/r/mosaicml/pytorch/tags)
+  * Recommended tag: `mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04`
+  * This image comes pre-configured with the following dependencies:
+    * PyTorch Version: 1.12.1
+    * CUDA Version: 11.6
+    * Python Version: 3.9
+    * Ubuntu Version: 20.04
+
+* Use a system with NVIDIA GPU(s)
+
+* Install requirements via: `pip install -r requirements.txt`
+
+* Prepare a local copy of the C4 dataset via instructions below.
+
+## Dataset preparation
+
+To run pre-training, you'll need to make yourself a local copy of the C4 pre-training dataset.
+If you only want to profile these BERTs, we recommend that you **only download and prepare the `val` split**,
```

Comment:
```suggestion
If you only want to profile the speed of these BERTs, we recommend that you **only download and prepare the `val` split**,
```
Not confident that people with zero perf background will get that "profile" always refers to timing, as opposed to just being a synonym for "characterize" or "analyze".

Reply:
Also, is this a common enough case to merit the space it takes to mention it?